rslib/src/backend/search/browser_table.rs

```diff
@@ -3,114 +3,88 @@
-    browser_table::Column::Question,
-    browser_table::Column::Answer,
-    browser_table::Column::CardDeck,
-    browser_table::Column::CardDue,
-    browser_table::Column::CardEase,
-    browser_table::Column::CardLapses,
-    browser_table::Column::CardInterval,
-    browser_table::Column::CardMod,
-    browser_table::Column::CardReps,
-    browser_table::Column::CardTemplate,
-    browser_table::Column::NoteCreation,
-    browser_table::Column::NoteField,
-    browser_table::Column::NoteMod,
-    browser_table::Column::NoteTags,
-    browser_table::Column::Notetype,
-];
+use strum::IntoEnumIterator;
 
-const NOTE_COLUMNS: [browser_table::Column; 11] = [
-    browser_table::Column::NoteCards,
-    browser_table::Column::NoteCreation,
-    browser_table::Column::NoteDue,
-    browser_table::Column::NoteEase,
-    browser_table::Column::NoteField,
-    browser_table::Column::NoteInterval,
-    browser_table::Column::NoteLapses,
-    browser_table::Column::NoteMod,
-    browser_table::Column::NoteReps,
-    browser_table::Column::NoteTags,
-    browser_table::Column::Notetype,
-];
+use crate::{backend_proto as pb, browser_table, collection::Collection, i18n::I18n};
 
 impl Collection {
-    pub(crate) fn all_browser_card_columns(&self) -> pb::BrowserColumns {
-        self.to_pb_columns(&CARD_COLUMNS)
-    }
-
-    pub(crate) fn all_browser_note_columns(&self) -> pb::BrowserColumns {
-        self.to_pb_columns(&NOTE_COLUMNS)
-    }
-
-    fn to_pb_columns(&self, columns: &[browser_table::Column]) -> pb::BrowserColumns {
-        let mut columns: Vec<pb::BrowserColumn> =
-            columns.iter().map(|c| c.to_pb_column(&self.tr)).collect();
+    pub(crate) fn all_browser_columns(&self) -> pb::BrowserColumns {
+        let mut columns: Vec<pb::browser_columns::Column> = browser_table::Column::iter()
+            .filter(|&c| c != browser_table::Column::Custom)
+            .map(|c| c.to_pb_column(&self.tr))
+            .collect();
         columns.sort_by(|c1, c2| c1.label.cmp(&c2.label));
         pb::BrowserColumns { columns }
     }
 }
 
 impl browser_table::Column {
-    fn to_pb_column(self, i18n: &I18n) -> pb::BrowserColumn {
-        pb::BrowserColumn {
+    fn to_pb_column(self, i18n: &I18n) -> pb::browser_columns::Column {
+        pb::browser_columns::Column {
             key: self.to_string(),
             label: self.localized_label(i18n),
-            is_sortable: self.is_sortable(),
-            sorts_reversed: self == browser_table::Column::NoteField,
+            notes_label: self.localized_notes_label(i18n),
+            sorting: self.sorting() as i32,
             uses_cell_font: self.uses_cell_font(),
-            aligns_centered: self.aligns_centered(),
+            alignment: self.alignment() as i32,
         }
     }
 
-    fn is_sortable(self) -> bool {
-        !matches!(self, Self::Question | Self::Answer | Self::Custom)
+    fn sorting(self) -> pb::browser_columns::Sorting {
```

Comment:
Still think the methods here apart from to_pb_column() would be better placed in src/browser_table.rs - otherwise looking good!

Reply:
No objections, I only wanted to finish the logic first. 🙂

Reply:
Apropos, should I define an internal struct then or keep using pb::BrowserColumns::Column?

Reply:
For simple enums, especially longer ones, I think it's fine to just reuse it. I'd use a separate Rust struct in cases like:

- one or more fields would be better represented with newtypes
- one or more fields have an enum (which gets turned into an int in protobuf)
- there are any embedded messages, since in protobuf they'll be marked optional, and that's inconvenient to deal with when the data is expected to exist

If multiple .rs files will be referencing the struct, you can "pub(crate) use pb::Something", to allow other files to import it from the file that exports it instead of from the proto file

Reply:
2 actually applies, but I don't find it that inconvenient because the pb crate lets us use the Rust enums and cast them into ints at the end.
3 also applies, but I guess it's fine because the field is `repeated`, so we have a Vec instead of an Option?

Put like that, `browser_table::{Row, Cell, Color, Font}` are all unnecessary.

Reply:
Yep, repeated fields are not a worry. 

This module is building up data that's only really useful to consume by frontend GUI code as far as I can tell? If other Rust methods needed to operate on the Column struct, then I'd suggest we at least keep a Rust version of that, building directly to the pb struct as you're currently doing LGTM